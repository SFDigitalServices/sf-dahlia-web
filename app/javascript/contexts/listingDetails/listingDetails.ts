interface ListingDetails {
  /**  */
  uid: string

  /**  */
  email: string

  /**  */
  created_at: Date

  /**  */
  updated_at: Date
}

interface ListingDetailsData {
  data: any
  success: boolean
}

export { ListingDetails, ListingDetailsData }
