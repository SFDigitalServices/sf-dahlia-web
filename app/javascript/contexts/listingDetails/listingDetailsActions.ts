import { createAction } from "typesafe-actions"
import { RailsUnit } from "../../api/types/rails/listings/RailsUnit"
import { RailsAmiChart } from "../../api/types/rails/listings/RailsAmiChart"

export enum ListingDetailsActions {
  StartFetchingUnits = "SetFetchingUnits",
  FinishFetchingUnits = "FinishFetchingUnits",
  SetUnits = "SetUnits",
  SetAmiCharts = "SetAmiCharts",
  StartFetchingAmiCharts = "SetFetchingAmiCharts",
  SetFetchedAmiCharts = "SetFetchedAmiCharts",
  SetFetchingAmiChartsError = "SetFetchingAmiChartsError",
  SetFetchingUnitsError = "SetFetchingUnitsError",
  FinishFetchingAmiCharts = "FinishFetchingAmiCharts",
}

export const startFetchingUnits = createAction(ListingDetailsActions.StartFetchingUnits)()
export const setUnits = createAction(ListingDetailsActions.SetUnits)()
export const setAmiCharts = createAction(ListingDetailsActions.SetAmiCharts)()
export const finishFetchingAmiCharts = createAction(ListingDetailsActions.FinishFetchingAmiCharts)<
  RailsAmiChart[]
>()
export const setFetchedAmiCharts = createAction(ListingDetailsActions.SetFetchedAmiCharts)()
export const startFetchingAmiCharts = createAction(ListingDetailsActions.StartFetchingAmiCharts)()
export const finishFetchingUnits = createAction(ListingDetailsActions.FinishFetchingUnits)<
  RailsUnit[]
>()
export const setFetchingUnitsError = createAction(
  ListingDetailsActions.SetFetchingUnitsError
)<Error>()
export const setFetchingAmiChartsError = createAction(
  ListingDetailsActions.SetFetchingAmiChartsError
)<Error>()
