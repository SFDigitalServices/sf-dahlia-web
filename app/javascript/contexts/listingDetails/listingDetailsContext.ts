import { createContext } from "react"
import RailsUnit from "../../api/types/rails/listings/RailsUnit"
import { RailsAmiChart, RailsAmiChartMetaData } from "../../api/types/rails/listings/RailsAmiChart"

export type ContextProps = {
  units: RailsUnit[]
  amiCharts: RailsAmiChart[]
  fetchedAmiCharts: boolean
  fetchingAmiCharts: boolean
  fetchedUnits: boolean
  fetchingUnits: boolean
  fetchUnits: (listingId: string) => void
  fetchAmiCharts: (chartsToFetch: RailsAmiChartMetaData[]) => void
  fetchingAmiChartsError: Error
  fetchingUnitsError: Error
}

export default createContext<Partial<ContextProps>>({})
