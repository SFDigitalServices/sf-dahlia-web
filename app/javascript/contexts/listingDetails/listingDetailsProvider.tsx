import React, { useReducer } from "react"
import ListingDetailsReducer from "./listingDetailsReducer"
import ListingDetailsContext, { ContextProps } from "./listingDetailsContext"
import { getUnits, getAmiCharts } from "../../api/listingApiService"
import { RailsUnit } from "../../api/types/rails/listings/RailsUnit"
import { RailsAmiChart, RailsAmiChartMetaData } from "../../api/types/rails/listings/RailsAmiChart"
import {
  finishFetchingUnits,
  finishFetchingAmiCharts,
  startFetchingAmiCharts,
  startFetchingUnits,
  setFetchingAmiChartsError,
  setFetchingUnitsError,
} from "./listingDetailsActions"

interface ListingDetailsProviderProps {
  children?: React.ReactNode
}

const ListingDetailsProvider = (props: ListingDetailsProviderProps) => {
  const [state, dispatch] = useReducer(ListingDetailsReducer, {
    units: [],
    amiCharts: [],
    fetchingUnits: false,
    fetchedUnits: false,
    fetchingAmiCharts: false,
    fetchedAmiCharts: false,
    fetchingAmiChartsError: null,
    fetchingUnitsError: null,
  })

  const contextValues: ContextProps = {
    units: state.units,
    amiCharts: state.amiCharts,
    fetchedAmiCharts: state.fetchedAmiCharts,
    fetchingAmiCharts: state.fetchingAmiCharts,
    fetchedUnits: state.fetchedUnits,
    fetchingUnits: state.fetchingUnits,
    fetchingAmiChartsError: state.fetchingAmiChartsError,
    fetchingUnitsError: state.fetchingUnitsError,
    fetchUnits: (listingId: string) => {
      dispatch(startFetchingUnits())
      getUnits(listingId)
        .then((units: RailsUnit[]) => {
          dispatch(finishFetchingUnits(units))
        })
        .catch((error) => {
          console.error(error)
          dispatch(setFetchingUnitsError(error))
        })
    },
    fetchAmiCharts: (chartsToFetch: RailsAmiChartMetaData[]) => {
      dispatch(startFetchingAmiCharts())
      getAmiCharts(chartsToFetch)
        .then((amiCharts: RailsAmiChart[]) => {
          dispatch(finishFetchingAmiCharts(amiCharts))
        })
        .catch((error) => {
          console.error(error)
          dispatch(setFetchingAmiChartsError(error))
        })
    },
  }

  return (
    <ListingDetailsContext.Provider value={contextValues}>
      {props.children}
    </ListingDetailsContext.Provider>
  )
}

export default ListingDetailsProvider
