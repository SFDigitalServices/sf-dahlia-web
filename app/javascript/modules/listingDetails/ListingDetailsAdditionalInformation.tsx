import React from "react"
import { LinkButton, ListingDetailItem, t } from "@bloom-housing/ui-components"
import { RailsListing } from "../listings/SharedHelpers"
import { TextTruncate } from "../../components/TextTruncate"
import { isHabitatListing, isSale } from "../../util/listingUtil"
import { stripMostTags } from "../../util/filterUtil"
import { getTranslatedString } from "../../util/languageUtil"

export interface ListingDetailsAdditionalInformationProps {
  listing: RailsListing
  imageSrc: string
}

export const ListingDetailsAdditionalInformation = ({
  listing,
  imageSrc,
}: ListingDetailsAdditionalInformationProps) => {
  const getCommissionString = () => {
    return listing.Realtor_Commission_Unit === "percent"
      ? t("listings.realtorCommissionPercentage", {
          percentage: listing.Realtor_Commission_Amount,
        })
      : `$${listing.Realtor_Commission_Amount.toLocaleString()}`
  }

  return (
    <ListingDetailItem
      imageAlt={""}
      imageSrc={imageSrc}
      title={t("listings.additionalInformation.header")}
      subtitle={t("listings.additionalInformation.subheader")}
    >
      <div className="listing-detail-panel">
        {listing.Listing_Other_Notes && (
          <div className="info-card bg-gray-100 border-0">
            <h3 className="text-serif-xl">{t("listings.specialNotes")}</h3>
            <TextTruncate
              className="primary-lighter-markup-link translate"
              buttonClassName="text-blue-700"
              text={stripMostTags(
                getTranslatedString(
                  listing.Listing_Other_Notes,
                  "Listing_Other_Notes",
                  listing.translations
                )
              )}
            />
          </div>
        )}
        {(!!listing.Required_Documents || isSale(listing)) && (
          <div className="info-card bg-gray-100 border-0">
            <h3 className="text-serif-xl">{t("listings.requiredDocuments")}</h3>
            <div className="text-xs">
              <TextTruncate
                className="primary-lighter-markup-link translate"
                buttonClassName="text-blue-700"
                text={stripMostTags(
                  getTranslatedString(
                    listing.Required_Documents,
                    "Required_Documents",
                    listing.translations
                  )
                )}
              />
            </div>
            {isSale(listing) && !isHabitatListing(listing) && (
              <div className="text-xs mt-4">
                <TextTruncate
                  className="primary-lighter-markup-link translate"
                  buttonClassName="text-blue-700"
                  text={t("listings.requiredDocumentsAfterApplying", {
                    url: "https://sfmohcd.org/after-homebuyer-lottery",
                  })}
                />
              </div>
            )}
          </div>
        )}
        {listing.Legal_Disclaimers && (
          <div className="info-card bg-gray-100 border-0">
            <h3 className="text-serif-xl">{t("listings.importantProgramRules")}</h3>
            <div className="text-xs">
              <TextTruncate
                text={stripMostTags(
                  getTranslatedString(
                    listing.Legal_Disclaimers,
                    "Legal_Disclaimers",
                    listing.translations
                  )
                )}
                className="primary-lighter-markup-link translate"
                buttonClassName="text-blue-500"
              />
            </div>
          </div>
        )}
        {listing.CC_and_R_URL && (
          <div className="info-card bg-gray-100 border-0">
            <h3 className="text-serif-xl">{t("listings.cc&r")}</h3>
            <div className="text-xs">
              <TextTruncate
                className="primary-lighter-markup-link translate"
                buttonClassName="text-blue-500"
                text={t("listings.cc&rDescription")}
              />
              <br />
              <LinkButton href={listing.CC_and_R_URL} className={"mt-4"} newTab={true}>
                {t("listings.downloadPdf")}
              </LinkButton>
            </div>
          </div>
        )}
        {isSale(listing) && (
          <div className="info-card bg-gray-100 border-0">
            <h3 className="text-serif-xl">{t("listings.realtorCommission")}</h3>
            <div className="text-xs">
              {listing.Allows_Realtor_Commission ? (
                <>
                  <span className={"font-bold"}>{t("listings.realtorCommissionHeader")}</span>
                  {getCommissionString()}
                </>
              ) : (
                t("listings.realtorCommissionNotEligible")
              )}
              {listing.Realtor_Commission_Info && (
                <div className={"mt-4"}>
                  <span className={"font-bold"}>{t("listings.realtorCommissionHowTo")}</span>
                  <span>
                    <TextTruncate
                      className="primary-lighter-markup-link translate"
                      buttonClassName="text-blue-500"
                      text={stripMostTags(
                        getTranslatedString(
                          listing.Realtor_Commission_Info,
                          "Realtor_Commission_Info",
                          listing.translations
                        )
                      )}
                    />
                  </span>
                </div>
              )}
            </div>
          </div>
        )}
        {/* TODO: implement once we've established needs for sales listings
              {listing.isSale && (
                <div className="info-card bg-gray-100 border-0">
                  <h3 className="text-serif-xl">For the Buyer's Realtor</h3>
                  {listing.Allows_Realtor_Commission ? (
                    display realtor_commission_header
                    realtorComissionMessage
                    {listing.Realtor_Commission_Info && realtor_commission_how_to}
                  ) : display realtor_commission_not_eligible message}
                </div>
              )}
            */}
        {listing.Repricing_Mechanism && (
          <div className="info-card bg-gray-100 border-0">
            <h3 className="text-serif-xl">{t("listings.rePricing")}</h3>
            <div className="text-xs">
              <TextTruncate
                className="primary-lighter-markup-link translate"
                buttonClassName="text-blue-500"
                text={stripMostTags(
                  getTranslatedString(
                    listing.Repricing_Mechanism,
                    "Repricing_Mechanism",
                    listing.translations
                  )
                )}
              />
            </div>
          </div>
        )}
      </div>
    </ListingDetailItem>
  )
}
