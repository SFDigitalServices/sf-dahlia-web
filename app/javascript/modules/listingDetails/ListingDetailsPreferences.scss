.preferences-list {
  --info-card-title-color: var(--bloom-color-gray-950);
  --subtitle-color: var(--bloom-color-gray-700);
  --description-color: var(--bloom-color-gray-700);
}
