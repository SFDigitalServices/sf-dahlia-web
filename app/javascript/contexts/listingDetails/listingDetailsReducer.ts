import { createReducer } from "typesafe-actions"
import { ListingDetailsActions } from "./listingDetailsActions"
import { RailsUnit } from "../../api/types/rails/listings/RailsUnit"
import { RailsAmiChart } from "../../api/types/rails/listings/RailsAmiChart"
import { AmiChartData } from "./listingDetailsContext"

type ListingDetailsState = {
  fetchingUnits: boolean
  fetchedUnits: boolean
  fetchingAmiCharts: boolean
  fetchedAmiCharts: boolean
  units: RailsUnit[]
  amiChartData: AmiChartData
  amiCharts: RailsAmiChart[]
  fetchingAmiChartsError: Error
  fetchingUnitsError: Error
}

const ListingDetailsReducer = createReducer({ units: [] } as ListingDetailsState, {
  [ListingDetailsActions.StartFetchingUnits]: (state) => {
    return { ...state, fetchingUnits: true }
  },
  [ListingDetailsActions.StartFetchingAmiCharts]: (state) => {
    return { ...state, fetchingAmiCharts: true }
  },
  [ListingDetailsActions.FinishFetchingUnits]: (state, { payload: units }) => {
    const amiChartYears: Array<number> = []
    const amiChartTypes = []
    const amiChartPercentages = []

    units.forEach((unit: RailsUnit) => {
      const amiChartYear = amiChartYears.find((year) => {
        return year === unit.AMI_chart_year
      })

      const amiChartType = amiChartTypes.find((type) => {
        return type === unit.AMI_chart_type
      })

      const amiChartPercentage = amiChartPercentages.find((type) => {
        return type === unit.Max_AMI_for_Qualifying_Unit
      })

      if (!amiChartYear) {
        amiChartYears.push(unit.AMI_chart_year)
      }

      if (!amiChartType) {
        amiChartTypes.push(unit.AMI_chart_type)
      }

      if (!amiChartPercentage) {
        amiChartPercentages.push(unit.Max_AMI_for_Qualifying_Unit)
      }
    })

    return {
      ...state,
      fetchingUnits: false,
      fetchedUnits: true,
      units,
      amiChartData: {
        years: amiChartYears,
        percentages: amiChartPercentages,
        types: amiChartTypes,
      },
    }
  },
  [ListingDetailsActions.FinishFetchingAmiCharts]: (state, { payload: amiCharts }) => {
    return {
      ...state,
      fetchingAmiCharts: false,
      fetchedAmiCharts: true,
      amiCharts: amiCharts?.map((amiChart: RailsAmiChart) => {
        return {
          ...amiChart,
          chartType: amiChart?.values[0]?.chartType,
        }
      }),
    }
  },
})

export default ListingDetailsReducer
