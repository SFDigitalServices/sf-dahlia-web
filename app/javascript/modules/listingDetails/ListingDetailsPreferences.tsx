import React, { useEffect, useState } from "react"
import { Icon, PreferencesList, t } from "@bloom-housing/ui-components"
import { RailsListingPreference } from "../../api/types/rails/listings/RailsListingPreferences"
import { PREFERENCES, PREFERENCES_IDS, PREFERENCES_WITH_PROOF } from "../constants"
import { getPreferences } from "../../api/listingApiService"
import "./ListingDetailsPreferences.scss"
import { getLocalizedPath } from "../../util/routeUtil"
import { getRoutePrefix, getSfGovUrl } from "../../util/languageUtil"
import { preferenceNameHasVeteran } from "../../util/listingUtil"

const determineDescription = (
  customPreferenceDescription: boolean,
  listingDescription: string,
  preferenceName: string,
  NRHPDistrict?: string
) => {
  if (customPreferenceDescription) {
    // todo: DAH-2767 Use Google Cloud Translate here
    return { description: listingDescription, descriptionClassName: "translate" }
  } else {
    return preferenceName === PREFERENCES.neighborhoodResidence && NRHPDistrict
      ? {
          description: t(
            "listings.lotteryPreference.Neighborhood Resident Housing Preference (NRHP).desc.withDistrict",
            { number: NRHPDistrict }
          ),
        }
      : {
          description: t(`listings.lotteryPreference.${preferenceName}.desc`),
        }
  }
}
export interface ListingDetailsPreferencesProps {
  listingID: string
}

export const mapPreferenceLink = (link: string): string => {
  switch (link) {
    case "http://sfmohcd.org/certificate-preference":
      return getSfGovUrl(link, 3275)
    case "http://sfmohcd.org/displaced-tenant-housing-preference":
      return getSfGovUrl(link, 7488)
    case "http://sfmohcd.org/housing-preference-programs":
      return getSfGovUrl(link, 3274)
    case "http://sfmohcd.org/neighborhood-resident-housing-preference":
      return getSfGovUrl(link, 3274)
    case "https://www.sf.gov/certain-buildings-have-special-lottery-preferences":
      return getSfGovUrl(link, 11931)
    default:
      return link
  }
}

export const ListingDetailsPreferences = ({ listingID }: ListingDetailsPreferencesProps) => {
  const [preferences, setPreferences] = useState<RailsListingPreference[]>([])

  useEffect(() => {
    void getPreferences(listingID).then((preferences) => {
      setPreferences(
        preferences?.filter((preference) => !preferenceNameHasVeteran(preference.preferenceName))
      )
    })
    return () => {
      setPreferences([])
    }
  }, [listingID])

  if (preferences?.length === 0) {
    return (
      <div className="flex justify-center">
        <Icon symbol="spinner" size="large" />
      </div>
    )
  }

  return (
    <>
      <PreferencesList
        listingPreferences={preferences?.map((preference, index) => {
          const links = []
          if (preference.readMoreUrl) {
            links.push({
              title: t("label.readMore"),
              url: mapPreferenceLink(preference.readMoreUrl),
              ariaLabel: t(`listings.lotteryPreference.${preference.preferenceName}.readMore`),
            })
          }
          if (PREFERENCES_WITH_PROOF.includes(preference.preferenceName)) {
            const routePrefix = getRoutePrefix(window.location.pathname)
            const anchorMap = {
              "Neighborhood Resident Housing Preference (NRHP)":
                PREFERENCES_IDS.neighborhoodResidence,
              "Rent Burdened / Assisted Housing Preference": PREFERENCES_IDS.assistedHousing,
              "Live or Work in San Francisco Preference": PREFERENCES_IDS.liveWorkInSf,
              "Alice Griffith Housing Development Resident": PREFERENCES_IDS.rightToReturn,
              [PREFERENCES.rightToReturnHuntersView]: PREFERENCES_IDS.rightToReturn,
              [PREFERENCES.rightToReturnPotrero]: PREFERENCES_IDS.rightToReturn,
            }
            links.push({
              title: t("label.viewDocumentChecklist"),
              ariaLabel: t(
                `listings.lotteryPreference.${preference.preferenceName}.additionalDocumentation`
              ),
              url:
                typeof routePrefix === "undefined"
                  ? `/document-checklist#${anchorMap[preference.preferenceName]}`
                  : getLocalizedPath(
                      `/document-checklist#${anchorMap[preference.preferenceName]}`,
                      routePrefix
                    ),
            })
          }

          return {
            ...determineDescription(
              preference.customPreferenceDescription,
              preference.description,
              preference.preferenceName,
              preference.NRHPDistrict
            ),
            links,
            ordinal: index + 1,
            subtitle:
              preference.unitsAvailable &&
              t("listings.lotteryPreference.upToUnits", preference.unitsAvailable),
            title: t(`listings.lotteryPreference.${preference.preferenceName}.title`),
          }
        })}
      />
    </>
  )
}
