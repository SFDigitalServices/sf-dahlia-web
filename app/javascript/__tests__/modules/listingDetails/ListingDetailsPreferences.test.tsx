import React from "react"
import { cleanup, waitFor } from "@testing-library/react"

import { ListingDetailsPreferences } from "../../../modules/listingDetails/ListingDetailsPreferences"
import { preferences as defaultPreferences } from "../../data/RailsListingPreferences/lottery-preferences-default"
import { preferences as sixPreferences } from "../../data/RailsListingPreferences/lottery-preferences-six"
import { renderAndLoadAsync } from "../../__util__/renderUtils"

const axios = require("axios")

jest.mock("axios")

describe("ListingDetailsPreferences", () => {
  afterEach(() => {
    cleanup()
    jest.clearAllMocks()
    jest.resetAllMocks()
  })

  it("display 3 default preferences - COP, DTHP, L/W", async () => {
    axios.get.mockResolvedValue({ data: { preferences: defaultPreferences } })

    const { asFragment, getByText } = await renderAndLoadAsync(
      <ListingDetailsPreferences listingID={"test"} />
    )

    await waitFor(() => getByText("Certificate of Preference (COP)"))
    expect(asFragment()).toMatchSnapshot()
  })

  it("display 6 preferences", async () => {
    axios.get.mockResolvedValue({ data: { preferences: sixPreferences } })

    const { asFragment, getByText } = await renderAndLoadAsync(
      <ListingDetailsPreferences listingID={"test"} />
    )

    await waitFor(() => getByText("Certificate of Preference (COP)"))
    expect(asFragment()).toMatchSnapshot()
  })
})
