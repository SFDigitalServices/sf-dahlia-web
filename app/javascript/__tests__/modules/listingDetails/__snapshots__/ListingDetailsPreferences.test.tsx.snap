// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingDetailsPreferences display 3 default preferences - COP, DTHP, L/W 1`] = `
<li
  className="list-section custom-counter__item"
>
  <header
    className="list-section__header custom-counter__header"
  >
    <hgroup>
      <h4
        className="custom-counter__title"
      >
        Lottery Preferences
      </h4>
      <span
        className="custom-counter__subtitle"
      >
        Anyone may enter the housing lottery for this listing. If your household has one of the following preferences, you will be considered in the order shown here. Each preference holder will be reviewed in lottery rank order.
      </span>
    </hgroup>
  </header>
  <div
    aria-live="assertive"
    role="alert"
  >
    <ol
      className="preferences-list"
    >
      <li
        className="preferences-list__item info-card"
      >
        <div
          className="preferences-list__number"
        >
          1
          <sup>
            st
          </sup>
        </div>
        <h4
          className="info-card__title"
        >
          Certificate of Preference (COP)
        </h4>
        <div
          className="preferences-list__subtitle"
        >
          Up to 6 units available
        </div>
        <div
          className="preferences-list__description"
        >
          For households in which at least one member holds a Certificate of Preference from the former San Francisco Redevelopment Agency. COP holders were displaced by Agency action generally during the 1960s and 1970s.
        </div>
        <div
          className="preferences-list__links"
        >
          <span>
            <a
              href="http://sfmohcd.org/certificate-preference"
              target="_blank"
            >
              Read More
            </a>
          </span>
        </div>
      </li>
      <li
        className="preferences-list__item info-card"
      >
        <div
          className="preferences-list__number"
        >
          2
          <sup>
            nd
          </sup>
        </div>
        <h4
          className="info-card__title"
        >
          Displaced Tenant Housing Preference (DTHP)
        </h4>
        <div
          className="preferences-list__subtitle"
        >
          Up to 1 units available
        </div>
        <div
          className="preferences-list__description"
        >
          For households in which at least one member holds a Displaced Tenant Housing Preference Certificate. DTHP Certificate holders are tenants who were evicted through either an Ellis Act Eviction or an Owner Move In Eviction, have been displaced by a fire, or who will experience an unaffordable rent increase due to affordability restrictions expiring. Once all units reserved for this preference are filled, remaining DTHP holders will receive Live/Work preference, regardless of their current residence or work location.
        </div>
        <div
          className="preferences-list__links"
        >
          <span>
            <a
              href="http://sfmohcd.org/displaced-tenant-housing-preference"
              target="_blank"
            >
              Read More
            </a>
          </span>
        </div>
      </li>
      <li
        className="preferences-list__item info-card"
      >
        <div
          className="preferences-list__number"
        >
          3
          <sup>
            rd
          </sup>
        </div>
        <h4
          className="info-card__title"
        >
          Live or Work in San Francisco Preference
        </h4>
        <div
          className="preferences-list__subtitle"
        >
          Up to 6 units available
        </div>
        <div
          className="preferences-list__description"
        >
          For households in which at least one member lives or works in San Francisco. Requires submission of proof. Please note in order to claim Work Preference, the applicant must currently work in San Francisco at least 75% of their working hours.
        </div>
        <div
          className="preferences-list__links"
        >
          <span>
            <a
              href="http://sfmohcd.org/housing-preference-programs"
              target="_blank"
            >
              Read More
            </a>
          </span>
          <span>
            <a
              href="/document-checklist"
              target="_blank"
            >
              View Document Checklist
            </a>
          </span>
        </div>
      </li>
    </ol>
    <p
      className="text-gray-700 text-tiny"
    >
      After all preference holders have been considered, any remaining units will be available to qualified applicants in lottery order.
    </p>
  </div>
</li>
`;

exports[`ListingDetailsPreferences display 6 preferences 1`] = `
<li
  className="list-section custom-counter__item"
>
  <header
    className="list-section__header custom-counter__header"
  >
    <hgroup>
      <h4
        className="custom-counter__title"
      >
        Lottery Preferences
      </h4>
      <span
        className="custom-counter__subtitle"
      >
        Anyone may enter the housing lottery for this listing. If your household has one of the following preferences, you will be considered in the order shown here. Each preference holder will be reviewed in lottery rank order.
      </span>
    </hgroup>
  </header>
  <div
    aria-live="assertive"
    role="alert"
  >
    <ol
      className="preferences-list"
    >
      <li
        className="preferences-list__item info-card"
      >
        <div
          className="preferences-list__number"
        >
          1
          <sup>
            st
          </sup>
        </div>
        <h4
          className="info-card__title"
        >
          Certificate of Preference (COP)
        </h4>
        <div
          className="preferences-list__subtitle"
        >
          Up to 1 units available
        </div>
        <div
          className="preferences-list__description"
        >
          For households in which at least one member holds a Certificate of Preference from the former San Francisco Redevelopment Agency. COP holders were displaced by Agency action generally during the 1960s and 1970s.
        </div>
        <div
          className="preferences-list__links"
        >
          <span>
            <a
              href="http://sfmohcd.org/certificate-preference"
              target="_blank"
            >
              Read More
            </a>
          </span>
        </div>
      </li>
      <li
        className="preferences-list__item info-card"
      >
        <div
          className="preferences-list__number"
        >
          2
          <sup>
            nd
          </sup>
        </div>
        <h4
          className="info-card__title"
        >
          Rent Burdened / Assisted Housing Preference
        </h4>
        <div
          className="preferences-list__subtitle"
        >
          Up to 2 units available
        </div>
        <div
          className="preferences-list__description"
        >
          For households who are currently paying more than 50% of income for housing costs or are living in public housing or project based Section 8 housing within San Francisco.
        </div>
        <div
          className="preferences-list__links"
        >
          <span>
            <a
              href="http://sfmohcd.org/housing-preference-programs"
              target="_blank"
            >
              Read More
            </a>
          </span>
          <span>
            <a
              href="/document-checklist"
              target="_blank"
            >
              View Document Checklist
            </a>
          </span>
        </div>
      </li>
      <li
        className="preferences-list__item info-card"
      >
        <div
          className="preferences-list__number"
        >
          3
          <sup>
            rd
          </sup>
        </div>
        <h4
          className="info-card__title"
        >
          Displaced Tenant Housing Preference (DTHP)
        </h4>
        <div
          className="preferences-list__subtitle"
        >
          Up to 3 units available
        </div>
        <div
          className="preferences-list__description"
        >
          For households in which at least one member holds a Displaced Tenant Housing Preference Certificate. DTHP Certificate holders are tenants who were evicted through either an Ellis Act Eviction or an Owner Move In Eviction, have been displaced by a fire, or who will experience an unaffordable rent increase due to affordability restrictions expiring. Once all units reserved for this preference are filled, remaining DTHP holders will receive Live/Work preference, regardless of their current residence or work location.
        </div>
        <div
          className="preferences-list__links"
        >
          <span>
            <a
              href="http://sfmohcd.org/displaced-tenant-housing-preference"
              target="_blank"
            >
              Read More
            </a>
          </span>
        </div>
      </li>
      <li
        className="preferences-list__item info-card"
      >
        <div
          className="preferences-list__number"
        >
          4
          <sup>
            th
          </sup>
        </div>
        <h4
          className="info-card__title"
        >
          Neighborhood Resident Housing Preference (NRHP)
        </h4>
        <div
          className="preferences-list__subtitle"
        >
          Up to 4 units available
        </div>
        <div
          className="preferences-list__description"
        >
          For households in which at least one member either lives within the Supervisor district of the project or within a half-mile of the project. Requires submission of proof of address.
        </div>
        <div
          className="preferences-list__links"
        >
          <span>
            <a
              href="http://sfmohcd.org/neighborhood-resident-housing-preference"
              target="_blank"
            >
              Read More
            </a>
          </span>
          <span>
            <a
              href="/document-checklist"
              target="_blank"
            >
              View Document Checklist
            </a>
          </span>
        </div>
      </li>
      <li
        className="preferences-list__item info-card"
      >
        <div
          className="preferences-list__number"
        >
          5
          <sup>
            th
          </sup>
        </div>
        <h4
          className="info-card__title"
        >
          Live or Work in San Francisco Preference
        </h4>
        <div
          className="preferences-list__subtitle"
        >
          Up to 5 units available
        </div>
        <div
          className="preferences-list__description"
        >
          For households in which at least one member lives or works in San Francisco. Requires submission of proof. Please note in order to claim Work Preference, the applicant must currently work in San Francisco at least 75% of their working hours.
        </div>
        <div
          className="preferences-list__links"
        >
          <span>
            <a
              href="http://sfmohcd.org/housing-preference-programs"
              target="_blank"
            >
              Read More
            </a>
          </span>
          <span>
            <a
              href="/document-checklist"
              target="_blank"
            >
              View Document Checklist
            </a>
          </span>
        </div>
      </li>
      <li
        className="preferences-list__item info-card"
      >
        <div
          className="preferences-list__number"
        >
          6
          <sup>
            th
          </sup>
        </div>
        <h4
          className="info-card__title"
        >
          Alice Griffith Housing Development Resident
        </h4>
        <div
          className="preferences-list__description"
        >
          For households in which at least one member was a resident of the Alice Griffith housing development. This includes baseline and current residents that lived in the targeted redevelopment site on or after the time of application for Choice Neighborhoods of October 26, 2010.
        </div>
        <div
          className="preferences-list__links"
        >
          <span>
            <a
              href="/document-checklist"
              target="_blank"
            >
              View Document Checklist
            </a>
          </span>
        </div>
      </li>
    </ol>
    <p
      className="text-gray-700 text-tiny"
    >
      After all preference holders have been considered, any remaining units will be available to qualified applicants in lottery order.
    </p>
  </div>
</li>
`;
